pkg/publish/default.go

```diff
@@ -120,18 +124,70 @@ func pushResult(tag name.Tag, br build.Result, opt []remote.Option) error {
 		return err
 	}
 
+	// writePeripherals implements walk.Fn
+	writePeripherals := func(_ context.Context, se oci.SignedEntity) error {
+		ociOpts := []ociremote.Option{ociremote.WithRemoteOptions(opt...)}
+
+		// Respect COSIGN_REPOSITORY
+		targetRepoOverride, err := ociremote.GetEnvTargetRepository()
+		if err != nil {
+			return err
+		}
+		if (targetRepoOverride != name.Repository{}) {
+			ociOpts = append(ociOpts, ociremote.WithTargetRepository(targetRepoOverride))
+		}
+		h, err := se.(interface{ Digest() (v1.Hash, error) }).Digest()
```

Comment:
Wut

Reply:
We don't have a `Digestable` interface, and this is what I've been using as a surrogate.  This has to be either `SignedImage` or `SignedImageIndex` and both implement this.

Reply:
Or maybe I'll implement `WriteSBOM` and just hide it from you 😂

Reply:
We'll only have SBOMs for images, so can we just cast that here?

This is a bit complicated since we'll want to sign the whole index, but presumably sign each platform-specific SBOM.

Reply:
This is called on both indices and images regardless of which we're dealing with.  For indices below this is handled by the `walk.SignedEntity`.  Ultimately it'll write everything it needs to (sigs for index, SBOMs for images), and we'll have flexibility to get smarter about placement without worrying too much.  I don't feel super strongly, but I'm not sure what exactly you are proposing as an alternative?